session/session_test.go

```diff
@@ -48,11 +52,60 @@ func TestReceiveEventWhenSubscribing(t *testing.T) {
 		if event != SessionDisconnected {
 			t.Error("Expected to receive disconnected: ", event)
 		}
+	case <-time.After(5 * time.Second):
+		t.Error("Failed to receive event")
+	}
 
-		if <-events != SessionReconnected {
+	select {
+	case event := <-events:
+		if event != SessionReconnected {
 			t.Error("Expected to receive reconnected: ", event)
 		}
-	case <-time.After(100 * time.Millisecond):
+	case <-time.After(5 * time.Second):
 		t.Error("Failed to receive event")
 	}
 }
+
+func TestResumeZKSessionWithValidSession(t *testing.T) {
+	proxy := CreateProxy(t)
+	defer proxy.Delete()
+
+	store, err := NewZKSession("localhost:27445", 200*time.Millisecond, nil)
+	if err != nil {
+		t.Error("Failed to connect to Zookeeper: ", err)
+	}
+
+	existingClientId, err := store.ClientId().Save()
+	if err != nil {
+		t.Error("Failed to save clientId: ", err)
+	}
+
+	store.Delete("localhost:27445", -1)
```

Comment:
When `store.Close()` is used the session can not be resumed. It continues to send the event `ZooKeeper connection closed`. From the documentation: 

> SessionClosed is normally only returned as a direct result of calling Close() on the ZKSession object. It is a terminal state; the connection will not be re-established.